Firestore/core/test/unit/local/leveldb_local_store_test.cc

```diff
@@ -46,6 +72,233 @@ INSTANTIATE_TEST_SUITE_P(LevelDbLocalStoreTest,
                          LocalStoreTest,
                          ::testing::Values(Factory));
 
+class LevelDbLocalStoreTest : public LocalStoreTestBase {
+ public:
+  LevelDbLocalStoreTest() : LocalStoreTestBase(Factory()) {
+  }
+};
+
+TEST_F(LevelDbLocalStoreTest, AddsIndexes) {
+  FieldIndex index_a = MakeFieldIndex("coll", 0, FieldIndex::InitialState(),
+                                      "a", model::Segment::Kind::kAscending);
+  FieldIndex index_b = MakeFieldIndex("coll", 1, FieldIndex::InitialState(),
+                                      "b", model::Segment::Kind::kDescending);
+  FieldIndex index_c =
+      FieldIndex{2,
+                 "coll",
+                 {model::Segment{Field("c1"), model::Segment::Kind::kAscending},
+                  model::Segment{Field("c2"), model::Segment::Kind::kContains}},
+                 model::FieldIndex::InitialState()};
+
+  ConfigureFieldIndexes({index_a, index_b});
+  auto field_indexes = GetFieldIndexes();
+  ASSERT_EQ(field_indexes, Vector(FieldIndex(index_b), FieldIndex(index_a)));
```

Comment:
Android's `assertThat(fieldIndexes).containsExactly(indexA, indexB);` works regardless of the ordering of elements. You can probably use a `std::set` here since `FieldIndex` can be ordered.

Otherwise this test might break sometime in the future when nothing is actually wrong.